src/client/mcp-client.ts

```diff
@@ -261,22 +259,105 @@ export class MCPClient implements ToolProvider {
      * @returns Array of available tools
      */
     async getTools(): Promise<ToolSet> {
+        const tools: ToolSet = {};
         try {
-            const response = await this.client.request(
-                { method: 'tools/list', params: {} },
-                ToolsListSchema
-            );
-            return response.tools.reduce<ToolSet>((acc, tool) => {
-                acc[tool.name] = {
-                    description: tool.description,
-                    parameters: tool.inputSchema,
-                };
-                return acc;
-            }, {});
+            // Call listTools with parameters only
+            const listToolResult = await this.client.listTools({});
+            logger.debug(`listTools result: ${JSON.stringify(listToolResult, null, 2)}`);
+
+            // Assume listToolResult.tools exists and iterate
+            if (listToolResult && listToolResult.tools) {
+                listToolResult.tools.forEach((tool: any) => {
+                    tools[tool.name] = {
+                        description: tool.description || 'No description provided',
+                        // Map inputSchema from SDK to parameters in local Tool type
+                        parameters: tool.inputSchema || { type: 'object', properties: {} },
```

Comment:
not sure if we should have defaults here for `description` and `parameters`, might lead to weird side effects that will be hard to debug. if the tool description or input schema isn't there it's probably better to have a hard failure rather than have tools act strangely.